projects/storefrontlib/src/cms-components/product/product-list/container/product-scroll/product-scroll.component.ts

```diff
@@ -94,6 +116,15 @@ export class ProductScrollComponent implements OnDestroy {
     }
     this.setConditions();
     this.ref.markForCheck();
+
+    if (
+      this.productListComponentService.latestScrollCriteria &&
+      this.model.pagination.currentPage <
+        this.productListComponentService.latestScrollCriteria.currentPage
+    ) {
+      // load next page to expend the list
+      this.loadNextPage(this.model.pagination.currentPage + 1);
```

Comment:
this gets called when:
- go to a plp and make sure it loads a new page
- click back
- click another category, so another plp page

this will fire again and the data will be messed up (it will show 0 products in breadcrumb and such)

- to fix is to handle if the query is not the same, so it won't fire or modyf loadNextPage so that it handles this case and no 'load next page'

Reply:
for different query, the `this.productListComponentService.latestScrollCriteria` will be `undefined`.